sdk/storage/azure-storage-file-share/src/test/java/com/azure/storage/file/share/FileAsyncApiTests.java

```diff
@@ -1209,109 +1177,116 @@ public void deleteFileError() {
 
     @Test
     public void deleteFileLease() {
-        primaryFileAsyncClient.create(1024).block();
-        String leaseId = createLeaseClient(primaryFileAsyncClient).acquireLease().block();
-
-        StepVerifier.create(primaryFileAsyncClient.deleteWithResponse(new ShareRequestConditions().setLeaseId(leaseId)))
+        StepVerifier
+            .create(primaryFileAsyncClient.create(1024)
+                .then(createLeaseClient(primaryFileAsyncClient).acquireLease())
+                .flatMap(leaseId -> primaryFileAsyncClient
+                    .deleteWithResponse(new ShareRequestConditions().setLeaseId(leaseId))))
             .assertNext(it -> FileShareTestHelper.assertResponseStatusCode(it, 202))
             .verifyComplete();
     }
 
     @Test
     public void deleteFileLeaseFail() {
-        primaryFileAsyncClient.create(1024).block();
-        createLeaseClient(primaryFileAsyncClient).acquireLease().block();
-
-        StepVerifier
-            .create(primaryFileAsyncClient
+        primaryFileAsyncClient.create(1024)
+            .then(createLeaseClient(primaryFileAsyncClient).acquireLease())
+            .flatMap(leaseId -> primaryFileAsyncClient
                 .deleteWithResponse(new ShareRequestConditions().setLeaseId(testResourceNamer.randomUuid())))
+            .as(StepVerifier::create)
             .verifyError(ShareStorageException.class);
```

Comment:
Although this works, I think it is important to follow the normal structure of StepVerifier we've been using for other tests unless absolutely necessary.